src/hooks/useClassNames.js

```diff
@@ -0,0 +1,7 @@
+import cx from 'classnames';
+
+function useClassNames(baseClassName, props = {}, conditionalClassNames = {}) {
+  return cx(baseClassName, props.className, conditionalClassNames);
+}
+
+export { useClassNames };
```

Comment:
🔴  this shouldn't be a hook, it's just a helper function to make the common className prop extending easier vs all the bloat of add prop types and the same code everywhere.

I will move this into a different DIR.